### PR TITLE
Enable use of override for session timeout in hours, fix captcha bypass bug.

### PR DESCRIPTION
**Pull request by alexellis**
- Enable use of override for session timeout. This is more useful than having to hard-code and rebuild the code for the previous 4 hour limit. Just set environmental variable and start the app.
- Future work may involve breaking down into minutes, but this is a good minimum delivery to provide value to end-user/developer.
    
- Fixes bug in Captcha code by introducing new landing page. This is not a new go template, it's a separate HTML file because SRP - single reponsibility principle. Happy for this to be refacted after merging commit.
    
- Fix for including Docker 1.12 override has been removed for later PR.

### CODE REVIEW

**Comment by alexellis** [comment]
@marcosnils this addresses your concerns, and I'm OK with having a separate HTML file for the bypass screen. If you want to refactor post-merge then by all means go ahead.

It also has the backwards compatibility feature temporarily removed.

**Comment by marcosnils** [comment]
@alexellis I still see some missing stuff from my previous comments on #48

https://github.com/franela/play-with-docker/pull/48#discussion_r88816468

**Comment by alexellis** [comment]
I've addressed your comments and added comments in the code etc. @marcosnils

**Comment by marcosnils** [comment]
> I've addressed your comments and added comments in the code etc

I think I'm missing something @alexellis. In this comment (https://github.com/franela/play-with-docker/pull/48#discussion_r88816468) I've suggested to change the return type to time.Duration and use ParseDuration function to parse the EXPIRY env variable. I don't see those changes reflected in this PR.

**Comment by alexellis** [comment]
Look at the `//` comments for more information. This is a minimal viable product - i.e. first iteration of the change to make the existing hard coded hours into a configurable value. Minutes etc can be done at a later date.

I'm not convinced that minute-level granularity is absolutely necessary for this to be useful.

**Comment by xetorthio** [comment]
Yeah. I think it make sense to keep this in hours for now.

**Comment by alexellis** [comment]
@xetorthio OK to merge then? @marcosnils said on Slack that you had something blocking this?

**Comment by alexellis** [comment]
@xetorthio this has been left open 5 days

**Comment by xetorthio** [comment]
Sorry for the delay, I was trying to check a different way to bypass recaptcha. But there is no other way than excluding it directly.
I am OK with this PR. I don't really mind having a different template with and without recaptcha.

File: www/assets/app.js [hunk 1]
```diff
@@ -3,6 +3,16 @@
 
 	var app = angular.module('DockerPlay', ['ngMaterial']);
 
+    // Automatically redirects user to a new session when bypassing captcha.
+    // Controller keeps code/logic separate from the HTML
+    app.controller("BypassController", ['$scope', '$log', '$http', '$location', '$timeout', '$mdDialog', '$window', function($scope, $log, $http, $location, $timeout, $mdDialog, $window) {
+        setTimeout(function() {
+            var el = document.querySelector("#submit");
+            console.log(el);
```
Comment (marcosnils) [inline www/assets/app.js]: I believe this print is unnecessary right?

**Comment by alexellis** [comment]
Rebase fixed, debug log removed. @marcosnils @xetorthio I'd appreciate it if you guys could merge because I've wasted time on two occasions having to rebase / merge.

File: www/assets/app.js (hunk 1 above)
Comment (alexellis) [inline www/assets/app.js]: Removed.

**Comment by marcosnils** [comment]
@alexellis don't worry, this PR has the highest priority. I'll test it out and merge

**Comment by marcosnils** [comment]
@alexellis fixed the build problem and now the angular app doesn't start

image

**Comment by alexellis** [comment]
I've fixed the "welcome" template issue which was caused by merge.. this has been quite a sticky PR since the code I've edited kept getting worked on over the last 10 days :-/

**Comment by marcosnils** [comment]
> Which page did you get the error on?

the main session page with recaptcha disabled. After I get redirected to the main page, I get that angular error.

**Comment by alexellis** [comment]
So what changes have gone in from your side to that page since it was working before? I can't actually test this since we binned the Docker 1.12 override.

**Comment by marcosnils** [comment]
@alexellis found the issue. You removed the angular-material module from the index.html page here < https://github.com/franela/play-with-docker/pull/51/files#diff-ab469c2203f00940178ab4d78eacaad6L105

Can you re-add it?

**Comment by marcosnils** [comment]
@alexellis should we add https in the `bypass.html` as well?.

**Comment by marcosnils** [comment]
Just tried locally and it works. Thx for the hard work @alexellis and sorry for the delay in merging this

**Pull request merged**